bin/varnishtest/tests/u00016.vtc

```diff
@@ -0,0 +1,56 @@
+varnishtest "varnishncsa -j json escaping"
+
+feature cmd "python3 -c 'import json'"
+
+server s1 -repeat 3 {
+	rxreq
+	txresp
+} -start
+
+varnish v1 -vcl+backend {} -start
+
+client c1 {
+	txreq -hdr "x-request-id: YES" -hdr "test:ascii"
+	rxresp
+	txreq -hdr "x-request-id: ∃" -hdr "test:unicode"
+	rxresp
+	txreq -hdr "test:null"
+	rxresp
+} -run
+
+delay 1
+
+# ASCII string with -j is valid JSON
+shell -match "^YES$" {
+	varnishncsa -n ${v1_name} -d -q 'ReqHeader:test eq "ascii"' -j \
+		-F '{"xrid": "%{X-REQUEST-ID}i"}' \
+	| python3 -c 'import sys, json; print(json.load(sys.stdin)["xrid"])'
```

Comment:
You don't need to break a shell line with `\` before a `|`, the `|` character itself is a natural line breaker:

```sh
#!/bin/sh

echo helloworld |
grep -o l
```

Reply:
Fixed